### PR TITLE
DCOS-39797: refactor(nodes): make table always 100% wide

### PR DESCRIPTION
**Pull request by juliangieseke**
This adds a new spacing column to strech the whole table to 100% width.

Adding a new column instead of just using the last one helps preventing sizing issues
when that last column hast left and right aligned content.

Closes https://jira.mesosphere.com/browse/DCOS-39797

## Testing

Resize window and verify that table always has 100% and content stays left.

## Trade-offs

None

## Dependencies

None

### CODE REVIEW

**Comment by GeorgiSTodorov** [comment]
It looks weird for me at full screen with 1920x1080 resolution:

screenshot from 2018-07-27 11-57-54

Is this intended and if not can we consider using % instead of pixels?

**Comment by GeorgiSTodorov** [comment]
In another PR https://github.com/dcos/dcos-ui/pull/3107, @DanielMSchmidt just pointed out a bug that makes the caret hidden for the `Memory` and `Disk` columns if they are too narrow.

**Comment by juliangieseke** [comment]
yeah, but please lets fix that in a different issue to prevent conflicts 🙇

File: plugins/nodes/src/js/components/NodesTable-new.tsx [hunk 1]
```diff
@@ -1,64 +1,77 @@
 import * as React from "react";
-import { Table, Column } from "@dcos/ui-kit";
+import { Table, Column } from "@dcos/ui-kit/dist/packages";
 
 import NodesList from "#SRC/js/structs/NodesList";
 import Node from "#SRC/js/structs/Node";
 
 import { SortableColumnHeader } from "ui-kit-stage/SortableColumnHeader";
-import { SortDirection } from "../types/SortDirection";
+import { SortDirection } from "#PLUGINS/nodes/src/js/types/SortDirection";
```
Comment (DanielMSchmidt) [inline plugins/nodes/src/js/components/NodesTable-new.tsx]: Was this your import plugin? IMHO relative paths are better to reference things within a plugin.

Comment (juliangieseke) [reply plugins/nodes/src/js/components/NodesTable-new.tsx]: Uh, I think so 🤔 cant remember changing them. shall I revert?

Comment (DanielMSchmidt) [reply plugins/nodes/src/js/components/NodesTable-new.tsx]: Would love to 😇

**Comment by juliangieseke** [comment]
@GeorgiSTodorov regarding your screenshot - this is expected, the columns aren't responsive for performance issues, so this is our best way to scale the table 🤷‍♂️

**Comment by DanielMSchmidt** [comment]
@juliangieseke Merge conflicts

**Comment by juliangieseke** [comment]
fixed conflicts.

Also, after chatting with @jaxesn about this approach last week and talking to Philipp today, I decided to stick to this column _for now_, in future we want to change the way our table is generating column sizes. a quick idea is drafted in https://jira.mesosphere.com/browse/DCOS-39147 - which is also stated as todo comment in all columnSizer functions 👍

**Comment by juliangieseke** [comment]
@DanielMSchmidt ^ 🙇

File: plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import CompositeState from "#SRC/js/structs/CompositeState";
 import QueryParamsMixin from "#SRC/js/mixins/QueryParamsMixin";
 import NodesList from "#SRC/js/structs/NodesList";
 
-import NodesTable from "../../../components/NodesTable";
+import NodesTable from "../../../components/NodesTable-new";
```
Comment (DanielMSchmidt) [inline plugins/nodes/src/js/pages/nodes/nodes-table/NodesTableContainer.js]: It's not the ticket to enable this, so let's please remove this line!

File: plugins/nodes/src/js/components/NodesTable-new.tsx [hunk 3]
```diff
@@ -1,64 +1,77 @@
...
 
 import {
   hostnameSorter,
   hostnameRenderer,
   hostnameSizer
-} from "../columns/NodesTableHostnameColumn";
+} from "#PLUGINS/nodes/src/js/columns/NodesTableHostnameColumn";
```
Comment (DanielMSchmidt) [inline plugins/nodes/src/js/components/NodesTable-new.tsx]: Again, let's please move this back to relative paths!

File: plugins/nodes/src/js/columns/NodesTableCPUBarColumn.tsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ import Node from "#SRC/js/structs/Node";
 // import { IWidthArgs as WidthArgs } from "@dcos/ui-kit/packages/table/components/Column";
 import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
 import ProgressBar from "#SRC/js/components/ProgressBar";
-import { Cell } from "@dcos/ui-kit";
+import { Cell } from "@dcos/ui-kit/dist/packages";
```
Comment (DanielMSchmidt) [inline plugins/nodes/src/js/columns/NodesTableCPUBarColumn.tsx]: Why is this necessary?

**Comment by juliangieseke** [comment]
@tnguyen0 @DanielMSchmidt fixed 👍

File: plugins/nodes/src/js/columns/NodesTableZoneColumn.tsx [hunk 5]
```diff
@@ -34,6 +34,6 @@ export function zoneSorter(data: Node[], sortDirection: SortDirection): Node[] {
   return sort(data, comparators, { reverse });
 }
 
-export function zoneSizer(args: WidthArgs): number {
-  return Math.max(125, args.width / args.totalColumns);
+export function zoneSizer(_args: WidthArgs): number {
+  return 200;
```
Comment (nLight) [inline plugins/nodes/src/js/columns/NodesTableZoneColumn.tsx]: What logic/screen size did we consider coming up with those numbers?

File: plugins/nodes/src/js/components/NodesTable.tsx [hunk 6]
```diff
@@ -332,6 +336,12 @@ export default class NodesTable extends React.Component<
           cellRenderer={gpuRenderer}
           width={gpuSizer}
         />
+
+        <Column
```
Comment (nLight) [inline plugins/nodes/src/js/components/NodesTable.tsx]: `spacer.gif` of 2018 :trollface:

**Comment by nLight** [comment]
The main question: is this really _the only_ way we can do the full width? Spacer column?

**Comment by juliangieseke** [comment]
I discussed it with @jaxesn @orlandohohmeier @Poltergeist and @DanielMSchmidt -> tl;dr it is the only reasonable solution _for now_. 
in future we need to improve it, but we're not sure which way is the best. I added some comments to https://jira.mesosphere.com/browse/DCOS-39147 and I suggest that we use that issue to continue the discussion 😞

File: plugins/nodes/src/js/columns/NodesTableZoneColumn.tsx (hunk 5 above)
Comment (juliangieseke) [reply plugins/nodes/src/js/columns/NodesTableZoneColumn.tsx]: for now they are just fixed numbers @lilysamimi gave me.

File: plugins/nodes/src/js/columns/NodesTableSpacingColumn.tsx [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import * as React from "react";
+import Node from "#SRC/js/structs/Node";
+
+// TODO: DCOS-39079
+// import { IWidthArgs as WidthArgs } from "@dcos/ui-kit/packages/table/components/Column";
+import { IWidthArgs as WidthArgs } from "#PLUGINS/nodes/src/js/types/IWidthArgs";
```
[the remark this answers is not in the document]
Comment (nLight) [reply plugins/nodes/src/js/columns/NodesTableSpacingColumn.tsx]: This remain valid @juliangieseke

File: plugins/nodes/src/js/columns/NodesTableZoneColumn.tsx (hunk 5 above)
Comment (nLight) [reply plugins/nodes/src/js/columns/NodesTableZoneColumn.tsx]: ok

File: plugins/nodes/src/js/components/NodesTable.tsx (hunk 6 above)
Comment (nLight) [reply plugins/nodes/src/js/components/NodesTable.tsx]: welp 🤷‍♂️

**Comment by juliangieseke** [comment]
@tnguyen0 @nLight conflicts & import fixed 👍
